test/exe/deploy_test.rb

```diff
@@ -0,0 +1,120 @@
+# frozen_string_literal: true
+require 'test_helper'
+require 'krane/cli/krane'
+
+class DeployTest < KubernetesDeploy::TestCase
+  def test_deploy_with_default_options
+    set_krane_deploy_expectations
+    krane_deploy!
+  end
+
+  def test_deploy_parses_global_timeout
+    set_krane_deploy_expectations(new_args: { max_watch_seconds: 10 })
+    krane_deploy!(flags: '--global-timeout 10s')
+    set_krane_deploy_expectations(new_args: { max_watch_seconds: 60**2 })
+    krane_deploy!(flags: '--global-timeout 1h')
+  end
+
+  def test_deploy_parses_selector
+    selector = KubernetesDeploy::LabelSelector.new('name' => 'web')
+    KubernetesDeploy::LabelSelector.expects(:parse).returns(selector)
+    set_krane_deploy_expectations(new_args: { selector: selector })
+    krane_deploy!(flags: '--selector name:web')
+  end
+
+  def test_deploy_parses_bindings
+    # FIXME: this require is only needed intermittently
```

Comment:
the test_helper.rb should require everything this is rather odd.

Reply:
For future reference: `test_helper.rb` requires `lib/kubernetes_deploy`, which requires everything else; `lib/kubernetes_deploy/bindings_parser`, however, is **not** required at the top level of any file inside `lib`, which is why it wasn't being loaded here. I sent a commit explaining a little bit about this and moving this `require` to the top of the test file.